tracer/src/Datadog.Trace/IAST/Source.cs

```diff
@@ -7,23 +7,38 @@
 
 namespace Datadog.Trace.Iast;
 
-internal readonly struct Source
+internal class Source
```

Comment:
It is used also when tainting headers and request parameters. The reason of this change is that we need to set the internal id after the creation of the Source because of the vulnerabilities json changes, so Source will not be immutable anymore.